modules/local/starfusion/build/main.nf

```diff
@@ -1,37 +1,32 @@
 process STARFUSION_BUILD {
-    tag 'star-fusion'
+    tag "$meta.id"
+    label 'process_high'
 
     conda "${moduleDir}/environment.yml"
-    container 'community.wave.seqera.io/library/dfam_hmmer_samtools_star-fusion_pruned:5694d82381bf039e'
+    container 'community.wave.seqera.io/library/dfam_hmmer_minimap2_samtools_pruned:63e3d21ca68ea531'
 
     input:
     tuple val(meta), path(fasta)
     tuple val(meta2), path(gtf)
+    path fusion_annot_lib
+    val dfam_species
 
     output:
-    path "*"  , emit: reference
+    path "ctat_genome_lib_build_dir"  , emit: reference
 
     script:
-    def binPath = (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1)  ? "prep_genome_lib.pl" : "/usr/local/src/STAR-Fusion/ctat-genome-lib-builder/prep_genome_lib.pl"
+    if (dfam_species != "human" && dfam_species != "mouse") {
+        error "Invalid species for --dfam_db. Only 'human' or 'mouse' are accepted. Provided: ${dfam_species}"
+    }
+    def args = task.ext.args ?: ''
     """
-    export TMPDIR=/tmp
-    wget http://ftp.ebi.ac.uk/pub/databases/Pfam/releases/Pfam34.0/Pfam-A.hmm.gz --no-check-certificate
-    wget https://github.com/FusionAnnotator/CTAT_HumanFusionLib/releases/download/v0.3.0/fusion_lib.Mar2021.dat.gz -O CTAT_HumanFusionLib_Mar2021.dat.gz --no-check-certificate
-    wget https://data.broadinstitute.org/Trinity/CTAT_RESOURCE_LIB/AnnotFilterRule.pm -O AnnotFilterRule.pm --no-check-certificate
-    wget https://www.dfam.org/releases/Dfam_3.4/infrastructure/dfamscan/homo_sapiens_dfam.hmm --no-check-certificate
-    wget https://www.dfam.org/releases/Dfam_3.4/infrastructure/dfamscan/homo_sapiens_dfam.hmm.h3f --no-check-certificate
-    wget https://www.dfam.org/releases/Dfam_3.4/infrastructure/dfamscan/homo_sapiens_dfam.hmm.h3i --no-check-certificate
-    wget https://www.dfam.org/releases/Dfam_3.4/infrastructure/dfamscan/homo_sapiens_dfam.hmm.h3m --no-check-certificate
-    wget https://www.dfam.org/releases/Dfam_3.4/infrastructure/dfamscan/homo_sapiens_dfam.hmm.h3p --no-check-certificate
-    gunzip Pfam-A.hmm.gz && hmmpress Pfam-A.hmm
-    $binPath \\
+    prep_genome_lib.pl \\
         --genome_fa $fasta \\
         --gtf $gtf \\
-        --annot_filter_rule AnnotFilterRule.pm \\
-        --fusion_annot_lib CTAT_HumanFusionLib_Mar2021.dat.gz \\
-        --pfam_db Pfam-A.hmm \\
-        --dfam_db homo_sapiens_dfam.hmm \\
-        --max_readlength $params.read_length \\
+        --dfam_db ${dfam_species} \\
+        --pfam_db current \\
+        --fusion_annot_lib $fusion_annot_lib \\
+        ${args} \\
```

Comment:
Can you write in the PR how and what you have tested this with? I will run my own build but it is good to follow the process of development

Reply:
I included how to run it in a comment, let me know if you can successfully run it with your build :)